app/group.go

```diff
@@ -173,8 +173,25 @@ func (a *App) GetGroupsByChannel(channelId string, page, perPage int) ([]*model.
 	return result.Data.([]*model.Group), nil
 }
 
-func (a *App) GetGroupsByTeam(teamId string, page, perPage int) ([]*model.Group, *model.AppError) {
-	result := <-a.Srv.Store.Group().GetGroupsByTeam(teamId, page, perPage)
+func (a *App) GetGroupsByTeam(teamId string, opts model.GroupSearchOpts) ([]*model.Group, int, *model.AppError) {
+	result := <-a.Srv.Store.Group().GetGroupsByTeam(teamId, opts)
+	if result.Err != nil {
+		return nil, 0, result.Err
+	}
+	groups := result.Data.([]*model.Group)
+
+	opts.PageOpts = nil
+	result = <-a.Srv.Store.Group().GetGroupsByTeam(teamId, opts)
```

Comment:
This looks inefficient, maybe a CountGroupsByTeam and let the database count the registers?

Reply:
@jespino Moved it to the DB tier in https://github.com/mattermost/mattermost-server/pull/10701/commits/e097e9a7e284e7869426548c6de2fc9464c12a94. For now the store method does both queries since it's always called together. In a later iteration I can refactor to split them into individually methods.

Reply:
Moved it into the store... going to split it into a new store method as part of this PR.